### ISSUE TITLE
Uneven labels lead to poor model training performance

### ISSUE THREAD

**Issue by xiaohangguo** [ultralytics/ultralytics#2231]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have found that when training my own dataset, uneven class labels result in poor model training performance, which can lead to a decrease in recall. Should I choose a lighter dataset or adopt some model training strategies.

The dataset I am using is a 'mini COCO' dataset with 80 categories, and I have added five self-defined categories.
![屏幕截图 2023-04-25 153320](https://user-images.githubusercontent.com/72799392/234206739-da6ca9e9-7c97-4886-8700-934498928be4.png)

**Comment by xiaohangguo** [comment]
The requirement and original intention is that I want the model to be able to detect both my customized dataset labels and my own dataset labels. However, due to the uneven number of sample labels, the trained model may not perform well. Do I need to remake the dataset or do I have a good training plan.
Additionally, I verified that I used the pre train (yolov8s, l, m...) model for fine-tuning on my own dataset. The trained model can only detect the dataset I defined, which was originally present on the pre trained model

**Comment by glenn-jocher** [comment]
@xiaohangguo it's common for uneven class distribution in a dataset to cause poor model training performance, which can lead to a decrease in recall. Instead of remaking the dataset, a better training plan could include strategies such as data augmentation, using class weights or oversampling the minority class. Regarding the pre-trained model, the fine-tuning process tends to maintain the original detection capabilities while adding new classes. To detect your customized dataset labels, you might need to train a new YOLOv8 model starting from scratch, then fine-tune it for your specific task. Nonetheless, I would suggest going through the YOLOv8 documentation and try experiments by yourself.

**Comment by xiaohangguo** [comment]
Yes, I have trained YOLOV8S, L, X and other models from scratch, but the pre trained model I obtained cannot detect labels in the original Coco dataset, such as person, car, cup, etc. What is the reason for this?

**Comment by glenn-jocher** [comment]
@xiaohangguo, The pre-trained YOLOv8 model is trained specifically on the COCO dataset and is generally useful for object detection tasks that involve objects that are similar to those in the COCO dataset. To detect labels in the original COCO dataset, it is ideal to use this pre-trained model directly, without further fine-tuning. It is important to ensure that the correct weight files are used, and the labels are correctly specified. Additionally, it is essential to set the confidence threshold correctly to optimize detection performance. If the model is not detecting objects as expected, it is worth reviewing and checking the configuration and adjusting the threshold.

**Comment by xiaohangguo** [comment]
@glenn-jocher 
Because I want to use a model that both detects COCO dataset labels and detects custom dataset labels. Can this be achieved?
I synthesized the dataset and then retrained it, but the effect was not good because of the imbalanced sample labels in my dataset.
